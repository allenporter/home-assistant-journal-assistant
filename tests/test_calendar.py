"""Test a calendar entity."""

import urllib
from http import HTTPStatus

from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant

from pytest_homeassistant_custom_component.typing import (
    ClientSessionGenerator,
)


@pytest.mark.usefixtures("setup_integration"
async def test_calendar(
    hass: HomeAssistant,
    hass_client: ClientSessionGenerator,
    snapshot: SnapshotAssertion,
) -> None:
    """Test a calendar entity."""

    state = hass.states.get("calendar.my_journal_daily")
    assert state is not None
    assert state.state == "off"
    assert dict(state.attributes) == {
        "friendly_name": "My Journal: Daily",
    }

    start = "2023-12-01T00:00:00Z"
    end = "2023-12-31T23:59:59Z"
    client = await hass_client()
    response = await client.get(
        f"/api/calendars/calendar.my_journal_daily?start={urllib.parse.quote(start)}&end={urllib.parse.quote(end)}"
    )
    assert response.status == HTTPStatus.OK
    resp = await response.json()
    assert [
        (event["start"], event["end"], event["summary"], event["description"])
        for event in resp
    ] == snapshot
