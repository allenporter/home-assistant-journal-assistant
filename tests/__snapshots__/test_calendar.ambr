# serializer version: 1
# name: test_calendar
  list([
    tuple(
      dict({
        'date': '2023-12-19',
      }),
      dict({
        'date': '2023-12-20',
      }),
      'Daily 2023-12-19',
      '''
        - (completed) migrate Dec to supernote
        - set Plans w/ mom
        - (migrated) call Dad re xmas plan
        - gifts info to mom & Sam
        - prod readiness pass
        - (migrated) xmas tree
        - Dot template: darker 5mm
      ''',
    ),
    tuple(
      dict({
        'date': '2023-12-20',
      }),
      dict({
        'date': '2023-12-21',
      }),
      'Daily 2023-12-20',
      '''
        - (migrated) xmas tree
        - text mom for xmas day
        - call Dad: xmas plans
        - (migrated) gifts plan w/ Cathrine
        - recognition vs old notes
        - living room clutter
        - okay of the house activity
        - (migrated) super note box clean up
        - flux-local helm magic
      ''',
    ),
    tuple(
      dict({
        'date': '2023-12-21',
      }),
      dict({
        'date': '2023-12-22',
      }),
      'Daily 2023-12-21',
      '''
        - cardboard breakdown
        - (migrated) Bowling w/ Q
        - (completed) flux-local helm
        - todo urls?
        - (migrated) gifts plan
        - windows xmas lights
        - (migrated) fitbit python
      ''',
    ),
    tuple(
      dict({
        'date': '2023-12-22',
      }),
      dict({
        'date': '2023-12-23',
      }),
      'Daily 2023-12-22',
      '- clean garage',
    ),
  ])
# ---
